### ISSUE TITLE
Unicode characters delete output completely when using lolcat

### ISSUE THREAD

**Issue by vypxl** [kovidgoyal/kitty#4919]
**Describe the bug**
When the input given to [lolcat](https://github.com/busyloop/lolcat) contains Unicode characters like `█`, its output is completely gone, even when piping it into a file or using tee. This could also be an issue with lolcat, but it only occurs in kitty, not in other terminal emulators or in tty.

**To Reproduce**
1. Start a kitty instance
2. Try one of the following commands:
 - `echo '█' | lolcat`
 - `echo '█' | lolcat -f > test.txt && cat test.txt`
 - `neofetch | lolcat` (assuming it displays some unicode characters in the logo)

Expected behavior: Colorful output.
Actual behavior: No output.

**Screenshots**
![2022-04-04-123249_1871x583_scrot](https://user-images.githubusercontent.com/13833431/161526537-039cd011-6bf4-487e-a866-f9fbd28fd55b.png)

**Environment details**
```
kitty 0.24.4 created by Kovid Goyal
Linux <> 5.16.14-1-MANJARO #1 SMP PREEMPT Fri Mar 11 14:12:18 UTC 2022 x86_64
Manjaro Linux 5.16.14-1-MANJARO  (<>) (/dev/tty)


DISTRIB_ID=ManjaroLinux
DISTRIB_RELEASE=21.2.5
DISTRIB_CODENAME=Qonos
DISTRIB_DESCRIPTION="Manjaro Linux"
Running under: X11
Frozen: False
Paths:
  kitty: /usr/bin/kitty
  base dir: /usr/lib/kitty
  extensions dir: /usr/lib/kitty/kitty
  system shell: /bin/bash

Config options different from defaults:

Important environment variables seen by the kitty process:
	PATH                                <>
	EDITOR                              vim
	SHELL                               /bin/bash
	DISPLAY                             :0
	USER                                <>
	XCURSOR_SIZE                        24
	XDG_SESSION_ID                      2
	XDG_SEAT_PATH                       /org/freedesktop/DisplayManager/Seat0
	XDG_SESSION_PATH                    /org/freedesktop/DisplayManager/Session1
	XDG_SESSION_CLASS                   user
	XDG_SEAT                            seat0
	XDG_DATA_DIRS                       /home/<>/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop
	XDG_SESSION_DESKTOP                 KDE
	XDG_CURRENT_DESKTOP                 KDE
	XDG_CONFIG_HOME                     /home/<>/.config
	XDG_SESSION_TYPE                    x11
	XDG_RUNTIME_DIR                     /run/user/1000
	XDG_VTNR                            1
	LC_PAPER                            en_US.UTF-8
	LC_TIME                             en_DE.UTF-8
	XDG_CONFIG_DIRS                     /home/<>/.config/kdedefaults:/etc/xdg
	LC_TELEPHONE                        en_US.UTF-8
	LC_ADDRESS                          en_US.UTF-8
	LC_NUMERIC                          en_US.UTF-8
	LC_NAME                             en_US.UTF-8
	LC_MONETARY                         en_DE.UTF-8
	LC_MEASUREMENT                      en_DE.UTF-8
	LC_IDENTIFICATION                   en_US.UTF-8

```

**Additional context**
Curiously, on my laptop, this does not happen. My desktop is running an RTX 2070 with proprietary nvidia drivers and an AMD 5900X. My laptop only has integrated intel graphics.

**Comment by kovidgoyal** [comment]
I cannot reproduce. And all lolcat does is change colors. Changing colors definitely works in kitty.
![screenshot](https://user-images.githubusercontent.com/1308621/161528954-4168a2eb-428c-4bfb-a8ec-b43459cb8680.png)

**Comment by vypxl** [comment]
I figured out what is happening:

For some reason, however I start kitty (even from other terminals which have $LANG set), $LANG is empty:

```
[094 14:17:16.426073] Failed to set locale with LANG: None
[094 14:17:16.426094] Failed to set locale with no LANG
```

This makes kitty fall back to US-ASCII, which causes lolcat to fail because it detects an invalid encoding for the unicode characters. Funnily, kitty still displays them correctly, even with the wrong encoding.

**Comment by kovidgoyal** [comment]
If the environment you are launching kitty from has LANG I dont see how its possible for kitty to not have LANG.

**Comment by vypxl** [comment]
Yeah that sure is weird.

I made another discovery: When launching kitty with a command from kde's `alt+f2` launcher, that command seems to have $LANG set: `kitty -c NONE --hold echo $LANG` outputs `en_US.UTF-8`.
However, if running any shell (even sh), it's not set: `kitty -c NONE --hold sh -c 'echo $LANG'` outputs nothing.

**Comment by kovidgoyal** [comment]
Then something in your shell's rc files is unsetting it.

**Comment by vypxl** [comment]
`kitty -c NONE sh --noprofile` also has $LANG unset, so there should be no script unsetting it?
Also, as I said, it only happens in kitty, not in Konsole, not in tty.

**Comment by vypxl** [comment]
Maybe the problem lies in pythons `locale`? Because if I try to run

```python
>>> import locale
>>> locale.setlocale(locale.LC_ALL, '')
```

as in kittys main script, it throws an error

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/locale.py", line 620, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```

On my laptop, this does not occur.
Seems to me like this is the real culprit, not kitty.

**Comment by kovidgoyal** [comment]
setlocale will fail if there is some locale related env var such as LC_*
or LANG or a couple of others that I dont remember now and the locale
deduced from them does not exist on your system

**Comment by vypxl** [comment]
I found out what the problem was. In my kde settings, I had set time and measurement locales to `en_DE.UTF-8`, which kde lists as an option, but which is not a valid locale. Changing that to `de_DE.UTF-8` solved the issues.
It seems like python `locale` just dies when some invalid locales are set..

As to why $LANG is unset, I think this indeed something kitty does, [here](https://github.com/kovidgoyal/kitty/blob/02e062af65ba8eb689821600e101807c33fa6f43/kitty/main.py#L341).
